File: factorial/factorial.py
# recursive approaches cannot handle larger inputs and will run into maximum
# recursion depth errors; iterative approaches are best for calculating
# the factorial for larger numbers


def factorial(x):
    y = 1
    for i in range(1, (x + 1)):
        y *= i
    return y

if __name__ == '__main__':
    import math
    for n in range(101):
        m = factorial(n)
        print n, ": ", m
        assert m == math.factorial(n)

File: factorial/factorial.c
// works for inputs up to 20

#include <stdio.h>

long int factorial(long int x){
    long int c = 1;
    long int o = 1;
    while (c <= x) {
        o *= c;
        c += 1;
    }
    return o;
}

int main(){
    long int n = 1;
    while ( n <= 20 ) {
        printf("%ld: %ld\n", n, factorial(n));
        n ++;
    }
}
